### PR TITLE
Upgrade to typescript 2.7 and get things passing

### PR DESCRIPTION
**Pull request by wadetandy**
This will be necessary to use the newest version of typedoc, which
relies on 2.7.x.  This is mostly just setting some existing internal
details to be looser than the default. Also locking typescript to 2.7.x
in package.json

### CODE REVIEW

File: src/util/write-payload.ts [hunk 1]
```diff
@@ -33,9 +33,9 @@ export class WritePayload<T extends JSORMBase> {
     const attrs: ModelRecord<T> = {}
 
     this._eachAttribute((key, value) => {
-      const snakeKey = underscore(key)
+      const snakeKey: keyof ModelRecord<T> = underscore(key) as any
 
-      if (!this.model.isPersisted || this.model.changes()[key]) {
+      if (!this.model.isPersisted || (<any>this.model.changes())[key]) {
```
Comment (richmolj) [inline src/util/write-payload.ts]: `any`?

File: src/util/write-payload.ts [hunk 2]
```diff
@@ -33,9 +33,9 @@ export class WritePayload<T extends JSORMBase> {
     const attrs: ModelRecord<T> = {}
 
     this._eachAttribute((key, value) => {
-      const snakeKey = underscore(key)
+      const snakeKey: keyof ModelRecord<T> = underscore(key) as any
```
Comment (richmolj) [inline src/util/write-payload.ts]: would it be better to put `string` instead of `as any` here?

Comment (wadetandy) [reply src/util/write-payload.ts]: need to convert to `any` to assign it back to the `keyof ModelRecord<T>`. It's a string before the `any` cast, and typescript doesn't like it.

File: src/util/write-payload.ts (hunk 1 above)
Comment (wadetandy) [reply src/util/write-payload.ts]: Something about typescript 2.7 doesn't like indexing into the `ModelRecord<T>` type we created, so need to cast to get it to behave.

**Comment by wadetandy** [comment]
closing this out for a bit of git housekeeping. will reopen in a while.

**Pull request closed without merging**